featurebyte/api/feature_list.py

```diff
@@ -1236,40 +1226,16 @@ def compute_historical_features(
         - [Feature.preview](/reference/featurebyte.api.feature.Feature.preview/):
           Preview feature group.
         """
-        payload = FeatureListGetHistoricalFeatures(
-            feature_list_id=self.id,
-            feature_clusters=self._get_feature_clusters(),
+        temp_historical_feature_table_name = f"__TEMPORARY_HISTORICAL_FEATURE_TABLE_{ObjectId()}"
+        temp_historical_feature_table = self.compute_historical_feature_table(
+            observation_table=observation_set,
+            historical_feature_table_name=temp_historical_feature_table_name,
             serving_names_mapping=serving_names_mapping,
         )
-
-        client = Configurations().get_client()
-        output = []
-        with alive_bar(
-            total=int(np.ceil(len(observation_set) / max_batch_size)),
-            title="Retrieving Historical Feature(s)",
-            **get_alive_bar_additional_params(),
-        ) as progress_bar:
-            for _, batch in observation_set.groupby(
-                np.arange(len(observation_set)) // max_batch_size
-            ):
-                response = client.post(
-                    "/feature_list/historical_features",
-                    data={"payload": payload.json()},
-                    files={"observation_set": dataframe_to_arrow_bytes(batch)},
-                )
-                if response.status_code != HTTPStatus.OK:
-                    raise RecordRetrievalException(
-                        response,
-                        resolution=(
-                            f"\nIf the error is related to connection broken, "
-                            f"try to use a smaller `max_batch_size` parameter (current value: {max_batch_size})."
-                        ),
-                    )
-
-                output.append(dataframe_from_arrow_stream(response.content))
-                progress_bar()  # pylint: disable=not-callable
```

Comment:
Yes, this progress bar is used for batching the observation set by rows via `max_batch_size` which is also removed. Batching the observation set that way was mainly to avoid http timeout on long running historical requests, but that is no longer an issue after switching to the async based workflow using HistoricalFeatureTable.

Reply:
Seems like `temp_historical_feature_table.to_pandas()` still showing progress bar (https://github.com/featurebyte/featurebyte/blob/97cbb4ff81c0868ab379866cb69f5e7cc4ad4f14/featurebyte/common/utils.py#L274-L304).

Reply:
Yes, the async task is also blocking. The SDK interface remains the same as before, except that now there is no risk of http request time out.